libs/designer-ui/src/lib/picker/filepickereditor.tsx

```diff
@@ -75,41 +107,51 @@ export const FilePickerEditor = ({
       <BaseEditor
         readonly={baseEditorProps.readonly}
         className="msla-filepicker-editor"
-        BasePlugins={{
-          tokens: baseEditorProps.BasePlugins?.tokens ?? true,
-        }}
-        initialValue={value}
+        BasePlugins={{ ...baseEditorProps.BasePlugins }}
+        initialValue={editorDisplayValue}
         onBlur={handleBlur}
         onFocus={baseEditorProps.onFocus}
         getTokenPicker={baseEditorProps.getTokenPicker}
         placeholder={baseEditorProps.placeholder}
         isTrigger={baseEditorProps.isTrigger}
         tokenPickerButtonEditorProps={{ showOnLeft: true }}
       >
-        <Change setValue={onValueChange} />
+        <PickerValueChange pickerDisplayValue={pickerDisplayValue} setEditorDisplayValue={setEditorDisplayValue} />
       </BaseEditor>
       <TooltipHost content={openFolderLabel} calloutProps={calloutProps} styles={hostStyles}>
-        <IconButton iconProps={folderIcon} aria-label={openFolderLabel} onClick={openTokenPicker} id={pickerIconId} disabled={true} />
+        <IconButton iconProps={folderIcon} aria-label={openFolderLabel} onClick={openFolderPicker} id={pickerIconId} />
       </TooltipHost>
       <Picker
         visible={showPicker}
         anchorId={pickerIconId}
         loadingFiles={isLoading}
-        currentPathSegments={getPathSegments(titleSegments)}
-        files={[]}
+        currentPathSegments={titleSegments}
+        files={filterItems(items, type, fileFilters)}
+        errorDetails={errorDetails}
         onCancel={() => setShowPicker(false)}
         handleFolderNavigation={onFolderNavigated}
+        handleItemSelected={onFileFolderSelected}
       />
     </div>
   );
 };
 
-const getPathSegments = (titleSegments?: PickerTitleInfo[]): IBreadcrumbItem[] => {
-  if (!titleSegments || titleSegments.length === 0) return [];
-  return titleSegments.map((titleSegment) => {
-    return {
-      text: titleSegment.title ?? '',
-      key: titleSegment.titleKey,
-    };
-  });
+const filterItems = (items?: TreeDynamicValue[], type?: string, fileFilters?: string[]): TreeDynamicValue[] => {
+  if (!items || items.length === 0) return [];
+  let returnItems = items;
+  if (type === PickerItemType.FOLDER) {
+    returnItems = items.filter((item) => item.isParent);
+  }
+  if (fileFilters && fileFilters.length > 0) {
+    returnItems = returnItems.filter((item) => {
+      return fileFilters.includes(item.mediaType ?? '');
```

Comment:
Can you change this to fileFilters.some(filter => equals(filter, item.mediaType))